File: 70-climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    typedef long long int ll;
    int climbStairs(int n) {
        // dp[n] => Total ways to reach the nTH destination 

        vector<ll>dp(n+1);

        dp[1]=1;

        if(n>1){
            dp[2]=2;
        }

        for(ll i=3;i<=n;i++){
            dp[i]=dp[i-1]+dp[i-2];
        }

        return (int)dp[n];
    }
};

File: 70-climbing-stairs/climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        // n Steps to reach the TOP 
        //climb 1 or 2 steps 
        // No.of Distinict ways, I can climb to TOP
        int[] dp = new int[n+1];
        dp[0]=1;dp[1]=1;
        for(int i=2;i<=n;i++){
            dp[i]=dp[i-1]+dp[i-2];
        }
        return dp[n];
    }
}